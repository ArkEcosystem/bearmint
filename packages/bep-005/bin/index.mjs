#!/usr/bin/env node
import { join } from 'node:path'

import { globby } from 'globby'
import meow from 'meow'

process.env.NODE_ENV = 'test'

const { flags } = meow({
	flags: {
		config: {
			type: 'string',
		},
	},
	importMeta: import.meta,
})

const { files } = (await import(join(process.cwd(), flags.config))).default
const tests = (await globby(files)).map((path) => join(process.cwd(), path))

for (const test of tests) {
	await import(test)
}
