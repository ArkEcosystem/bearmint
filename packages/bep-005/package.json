{
	"name": "@bearmint/bep-005",
	"version": "0.0.0",
	"private": true,
	"description": "This library is the official reference implementation for BEP-005 (https://bep.dev/005)",
	"keywords": [
		"bearmint",
		"tendermint",
		"blockchain",
		"framework",
		"modular"
	],
	"homepage": "https://github.com/bearmint/bearmint",
	"repository": {
		"type": "git",
		"url": "git+https://github.com/bearmint/bearmint.git"
	},
	"funding": "https://funding.bearmint.com/",
	"license": "MIT",
	"author": "Bearmint Mining Company <hello@bearmint.com> (https://bearmint.com/)",
	"type": "module",
	"exports": "./distribution/index.js",
	"types": "./distribution/index.d.ts",
	"bin": {
		"roadrunner": "./bin/index.mjs"
	},
	"files": [
		"bin",
		"distribution"
	],
	"scripts": {
		"audit": "pnpm audit --fix --prod",
		"build": "pnpm run clean && tsc",
		"build:watch": "pnpm run clean && tsc -w",
		"clean": "rm -rf distribution",
		"depcheck": "depcheck --ignores=@bearmint/bep-004,@bearmint/bep-005,@bearmint/bep-031,@bearmint/bep-039,globby,meow",
		"deps:major": "ncu -u --target=latest",
		"deps:minor": "ncu -u --target=minor",
		"deps:patch": "ncu -u --target=patch"
	},
	"dependencies": {
		"@bearmint/bep-013": "workspace:^",
		"@bearmint/bep-018": "workspace:^",
		"@bearmint/bep-116": "workspace:^",
		"@sinonjs/fake-timers": "9.1.2",
		"ansi-styles": "6.2.1",
		"concordance": "5.0.4",
		"globby": "13.1.2",
		"lodash.get": "4.4.2",
		"lodash.has": "4.5.2",
		"meow": "^11.0.0",
		"nock": "13.2.9",
		"pretty-ms": "8.0.0",
		"sinon": "14.0.1",
		"strip-ansi": "7.0.1",
		"vitest": "^0.24.5"
	},
	"devDependencies": {
		"@types/node": "18.11.9",
		"@types/sinon": "10.0.13",
		"@types/sinonjs__fake-timers": "8.1.2"
	},
	"engines": {
		"node": ">=18",
		"pnpm": ">=7.9.0"
	},
	"publishConfig": {
		"access": "restricted"
	}
}
