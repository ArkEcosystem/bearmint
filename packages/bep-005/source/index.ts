export * from './expect.js'
export * from './fake.js'
export * from './nock.js'
export * from './spy.js'
export * from './stub.js'
