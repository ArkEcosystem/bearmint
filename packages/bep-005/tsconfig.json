{
	"compilerOptions": {
		"typeRoots": ["node_modules/@types"],
		"outDir": "distribution"
	},
	"exclude": ["node_modules", "test/**/**.ts"],
	"extends": "../../config/tsconfig.json",
	"include": ["source/**/**.ts"]
}
