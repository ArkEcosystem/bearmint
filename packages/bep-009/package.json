{
	"name": "@bearmint/bep-009",
	"version": "0.0.0",
	"description": "This library is the official reference implementation for BEP-009 (https://bep.dev/009)",
	"keywords": [
		"bearmint",
		"tendermint",
		"blockchain",
		"framework",
		"modular"
	],
	"homepage": "https://github.com/bearmint/bearmint",
	"repository": {
		"type": "git",
		"url": "git+https://github.com/bearmint/bearmint.git"
	},
	"funding": "https://funding.bearmint.com/",
	"license": "MIT",
	"author": "Bearmint Mining Company <hello@bearmint.com> (https://bearmint.com/)",
	"type": "module",
	"exports": "./distribution/index.js",
	"types": "./distribution/index.d.ts",
	"files": [
		"distribution"
	],
	"scripts": {
		"audit": "pnpm audit --fix --prod",
		"build": "pnpm run clean && tsc",
		"build:watch": "pnpm run clean && tsc -w",
		"clean": "rm -rf distribution",
		"depcheck": "depcheck --ignores=@bearmint/bep-004,@bearmint/bep-005,@bearmint/bep-031,@bearmint/bep-039",
		"deps:major": "ncu -u --target=latest",
		"deps:minor": "ncu -u --target=minor",
		"deps:patch": "ncu -u --target=patch",
		"docs": "typedoc source",
		"knip": "knip",
		"lint": "eslint 'source/**/*.ts' --format=codeframe --config=../../config/eslintrc.js --fix",
		"test": "vitest run",
		"test:coverage": "vitest run --coverage",
		"test:ui": "vitest --ui",
		"test:watch": "vitest"
	},
	"dependencies": {
		"@bearmint/bep-013": "workspace:^",
		"@ethereumjs/rlp": "^4.0.0",
		"@noble/hashes": "1.1.3",
		"camelcase": "^7.0.0",
		"camelcase-keys": "^8.0.2",
		"dayjs": "1.11.6",
		"dot-prop": "^7.2.0",
		"env-paths": "^3.0.0",
		"matcher": "^5.0.0",
		"moize": "6.1.3",
		"semver": "7.3.8",
		"snake-case": "^3.0.4",
		"super-regex": "^0.2.0"
	},
	"devDependencies": {
		"@bearmint/bep-005": "workspace:^",
		"@types/node": "18.11.9",
		"@types/semver": "7.3.13"
	},
	"engines": {
		"node": ">=18",
		"pnpm": ">=7.9.0"
	},
	"publishConfig": {
		"access": "public"
	}
}
