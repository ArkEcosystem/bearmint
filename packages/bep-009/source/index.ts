export * from './assert.js'
export * from './base64.js'
export * from './big-number.js'
export * from './bytes.js'
export * from './camel-case.js'
export * from './directory.js'
export * from './dot-property.js'
export * from './environment-paths.js'
export * from './file.js'
export * from './glob.js'
export * from './hash.js'
export * from './hex.js'
export * from './is.js'
export * from './json.js'
export * from './non-nullable-map.js'
export * from './number.js'
export * from './numbers.js'
export * from './package.js'
export * as Paths from './paths.js'
export * from './regex.js'
export * as RLP from './rlp.js'
export * from './semver.js'
export * from './snake-case.js'
export * from './string.js'
export * from './uniq.js'
export * from './zeit.js'
