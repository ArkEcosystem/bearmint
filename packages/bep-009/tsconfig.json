{
	"compilerOptions": {
		"outDir": "distribution",
		"typeRoots": ["node_modules/@types"]
	},
	"exclude": ["node_modules", "test/**/**.ts"],
	"extends": "../../config/tsconfig.json",
	"include": ["source/**/**.ts"]
}
