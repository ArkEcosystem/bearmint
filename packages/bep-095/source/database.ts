import { ClassicLevel } from 'classic-level'

import type { MuSig } from './types.js'

export interface Signature {
	id: string
	publicKey: string
	signature: string
}

export interface StoreTxRaw {
	bytes: string
	muSig: MuSig
	signatures: Array<{ id: string; publicKey: string; signature: string }>
}

export interface StoreTx {
	hash: string
	muSig: MuSig
	signatures: Array<{ id: string; publicKey: string; signature: string }>
	timestamp: number
}

export function makeDatabase(path: string) {
	const database = new ClassicLevel(path, {
		valueEncoding: 'buffer',
	})

	return {
		async all() {
			const result: StoreTx[] = []

			for (const key of await database.keys().all()) {
				result.push(JSON.parse(await database.get(key)))
			}

			return result
		},
		async close() {
			await database.close()
		},
		async flush() {
			for (const key of await database.keys().all()) {
				await database.del(key)
			}
		},
		async forget(hash: string) {
			await database.del(hash)
		},
		async store(entry: StoreTx) {
			await database.put(entry.hash, JSON.stringify(entry))
		},
	}
}
