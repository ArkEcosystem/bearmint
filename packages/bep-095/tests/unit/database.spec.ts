import { expect, describe, it, beforeEach } from 'vitest'
import { extendExpect } from '@bearmint/bep-005'
extendExpect(expect)
import { bytesToHex, dirname } from '@bearmint/bep-009'
import { SignedTx } from '@bearmint/bep-018'

import { makeDatabase } from '../../source/database.js'

describe('Database', () => {
	it('should get all of the entries', async () => {
		const store = makeDatabase(`${dirname()}/level`)
		await store.flush()

		expect(await store.all()).toMatchSnapshot()

		await store.store({
			hash: bytesToHex(
				new SignedTx({
					gas: BigInt(1),
					hash: 'hash',
					message: {
						content: Buffer.alloc(0),
						handler: 'handler',
						network: 'network',
						version: 'version',
					},
					network: '134a3ccc33350717a7db79a93b5516ea0a5d4db0a60f03a08335e0c4c975d15e',
					nonce: BigInt(1),
					recipient: 'recipient',
					sender: 'sender',
					signature:
						'458561a00f87f53fa594303bccc3eaca6c09b38b091807e33a9ef53000ce31ca818b9509d37d055479a0038e06142d70fa83541a8d1d3589035c339491f2fa11',
					version: '1.0.0',
				}).toBinary(),
			),
			muSig: {
				m: 2,
				master: 'master',
				n: 3,
				slaves: ['1', '2', '3'],
			},
			signatures: [],
		})

		expect(await store.all()).toMatchSnapshot()

		await store.forget(
			'0a04686173681240313334613363636333333335303731376137646237396139336235353136656130613564346462306136306630336130383333356530633463393735643135651801220673656e6465722a09726563697069656e743001380142800134353835363161303066383766353366613539343330336263636333656163613663303962333862303931383037653333613965663533303030636533316361383138623935303964333764303535343739613030333865303631343264373066613833353431613864316433353839303335633333393439316632666131314a80013435383536316130306638376635336661353934333033626363633365616361366330396233386230393138303765333361396566353330303063653331636138313862393530396433376430353534373961303033386530363134326437306661383335343161386431643335383930333563333339343931663266613131',
		)

		expect(await store.all()).toMatchSnapshot()

		await store.close()
	})
})
