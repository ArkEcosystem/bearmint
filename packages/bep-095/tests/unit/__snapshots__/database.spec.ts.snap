// Vitest Snapshot v1

exports[`Database > should get all of the entries 1`] = `[]`;

exports[`Database > should get all of the entries 2`] = `
[
  {
    "hash": "0a05312e302e30120673656e6465721a09726563697069656e7420012801321b0a0768616e646c6572120776657273696f6e1a076e6574776f726b",
    "muSig": {
      "m": 2,
      "master": "master",
      "n": 3,
      "slaves": [
        "1",
        "2",
        "3",
      ],
    },
    "signatures": [],
  },
]
`;

exports[`Database > should get all of the entries 3`] = `
[
  {
    "hash": "0a05312e302e30120673656e6465721a09726563697069656e7420012801321b0a0768616e646c6572120776657273696f6e1a076e6574776f726b",
    "muSig": {
      "m": 2,
      "master": "master",
      "n": 3,
      "slaves": [
        "1",
        "2",
        "3",
      ],
    },
    "signatures": [],
  },
]
`;
